src/lib/utils/translations.ts

```diff
@@ -210,3 +212,19 @@ const getRequiredNamespacesForLayout = (layout?: string) => {
 
   return requiredNamespaces
 }
+
+/**
+ * Checks if the primary namespace .json file exists for the given locale
+ * @param locale Path locale
+ * @param primaryNs Primary namespace for the page
+ * @param existsSync Passed from fs library
+ * @returns false if the primary namespace .json file exists, true otherwise
+ */
+export const getIsContentNotTranslated = (
+  locale: string,
+  primaryNs: string,
+  existsSync: (arg0: string) => boolean
```

Comment:
This is implementation detail. No need to receive this function as an argument.

Consumer only cares about `locale` and `namespace`. The function should take care of the rest.

```suggestion
```

Reply:
Agree... Had done it this way initially to avoid importing from `fs` inside the `translations.ts` file, which is consumed by other non-page components that won't have access to `fs`:

<img width="611" alt="image" src="https://github.com/ethereum/ethereum-org-fork/assets/54227730/878a3f16-85a8-485e-9b8e-114101bbae61">

I think switching this to a dedicated util file would be the best way to go. That will avoid this conflict, and avoids passing a prop in this manner.